scheduler/src/main/java/org/pentaho/platform/scheduler2/quartz/BlockingQuartzJob.java

```diff
@@ -41,9 +42,17 @@
  */
 public class BlockingQuartzJob implements Job {
   public void execute( final JobExecutionContext jobExecutionContext ) throws JobExecutionException {
+    JobDataMap jobDataMap = null;
+    if ( jobExecutionContext.getJobDetail() != null && jobExecutionContext.getJobDetail().getJobDataMap() != null ) {
+      jobDataMap = jobExecutionContext.getJobDetail().getJobDataMap();
+    }
```

Comment:
Did you confirm that `jobExecutionContext` cannot be null? Although it makes sense to me to be the case, a test existed in the original code, and just wanna check.

Reply:
It can't be null.  I studied this carefully by looking at existing code that also uses jobExecutionContext without null checks.